src/Model/ModelManagerInterface.php

```diff
@@ -69,6 +71,16 @@ public function delete($object);
      */
     public function findBy($class, array $criteria = []);
 
+    // NEXT_MAJOR: Uncomment this method.
+    // /**
+    //  * @return object[]&Collection all objects matching the given identifiers.
+    //  *
+    //  * @phpstan-template T of object
+    //  * @phpstan-param class-string<T> $class
+    //  * @phpstan-return T[]
+    //  */
```

Comment:
```suggestion
    // /**
    //  * @return Collection<object>&object[] all objects matching the given identifiers.
    //  *
    //  * @phpstan-template T of object
    //  * @phpstan-param class-string<T> $class
    //  * @phpstan-return Collection<array-key, T>
    //  */
```

If we keep `Collection` as return type